methods.py

```diff
@@ -815,21 +815,20 @@ def get_compiler_version(env):
         "apple_patch3": -1,
     }
 
-    if not env.msvc:
-        # Not using -dumpversion as some GCC distros only return major, and
-        # Clang used to return hardcoded 4.2.1: # https://reviews.llvm.org/D56803
-        try:
-            version = (
-                subprocess.check_output([env.subst(env["CXX"]), "--version"], shell=(os.name == "nt"))
-                .strip()
-                .decode("utf-8")
-            )
-        except (subprocess.CalledProcessError, OSError):
-            print_warning("Couldn't parse CXX environment variable to infer compiler version.")
-            return ret
-    else:
+    if env.msvc and not using_clang(env):
         # TODO: Implement for MSVC
         return ret
+
```

Comment:
Can this be changed so that the diff is smaller? Eg. if you keep the if as is and on the else you make elif not using_clang(env)..

Reply:
If this was not cherry picked from upstream, making the diff smaller would be fine.

Reply:
Right, my bad, forgot it's cherry picked